File: kotlin-microservice/template/src/main/kotlin/{{tld}}/{{author}}/{{app_name}}/messaging/SampleMessageHandler.kt
package {{tld}}.{{author}}.{{app_name}}.messaging

import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class SampleMessageHandler : MessageHandler {
    private val log = LoggerFactory.getLogger(javaClass)

    override fun handle(body: String) {
        log.info("Received SQS message: {}", body)
    }
}

File: nodejs-typescript/template/src/messaging/sample-message.handler.ts
import { Injectable, Logger } from '@nestjs/common';

@Injectable()
export class SampleMessageHandler {
  private readonly logger = new Logger(SampleMessageHandler.name);

  async handle(body: string): Promise<void> {
    this.logger.log({ body }, 'Received SQS message');
  }
}
